### PR TITLE
Infra support for multiple targets in REST application deployment utility

### PR DESCRIPTION
**Pull request by anpanigr**
Enhanced the DeployUtil.deployUsingRest() to support multiple targets, instead of single cluster target.
Updated few Test Classes to use this enhanced utility method.

Added two convenience methods to deploy to individual cluster or server 
    deployToClusterUsingRest()
    deployToServerUsingRest()

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/3556  ( all passed ) 
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/3564  ( 1 intermittent failure not related to the current change)

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DeployUtil.java [hunk 1]
```diff
@@ -227,21 +227,24 @@ private static void createDeployJob(String deployScriptConfigMap, String namespa
 
   }
 
-
   /**
-   * Deploy application to a cluster using REST API with curl utility.
+   * Deploy application to a set of target using REST API with curl utility.
+   * Note the targets parameter should be a string with following format
+   *  { identity: [ clusters, 'cluster-1' ] }   for cluster target
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DeployUtil.java:233]: can you document how to deploy to multiple clusters and servers?

Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DeployUtil.java]: Updated the javadoc as follows  ...

Deploy application to a set of target using REST API with curl utility.
Note the targets parameter should be a string with following format
  {identity: [clusters,'cluster-1']}   for cluster target 
  {identity: [servers, 'admin-server']} for server target 
  OR a combination for multiple targets, for example 
   {identity: [clusters,'mycluster']}, {identity: [servers,'admin-server']}

**Review by sankarpn — APPROVED** [review]
LGTM

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DeployUtil.java [hunk 2]
```diff
@@ -236,11 +236,11 @@ private static void createDeployJob(String deployScriptConfigMap, String namespa
    * @param password admin server password
    * @param cluster name of the cluster to deploy application
    * @param archivePath local path of the application archive
-   * @param hostHeader name of the cluster to deploy application
+   * @param hostHeader Host header for the curl command 
    * @param appName name of the application
    * @return ExecResult 
    */
-  public static ExecResult deployUsingRest(String host, String port,
+  public static ExecResult deployToClusterUsingRest(String host, String port,
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DeployUtil.java:243]: do we need this method deployToClusterUsingRest as there is deployUsingRest which can take cluster or server or combination for targets? If deployToClusterUsingRest is used, can that be modified to use deployUsingRest and remove deployToClusterUsingRest method?

**Review by vanajamukkara — CHANGES_REQUESTED** [review]

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DeployUtil.java (hunk 2 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DeployUtil.java:243]: I kept the method deployToClusterUsingRest as convenience method with simple target syntax ( as cluster name )  as most of the IT class deploy the application to cluster.

Comment (vanajamukkara) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DeployUtil.java:243]: ok, can you refactor the code in these methods to remove duplicate code.

Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DeployUtil.java:242]: refactored the code  as follows  ...
 public static ExecResult deployToClusterUsingRest(String host, String port,
            String userName, String password, String cluster,
            Path archivePath, String hostHeader, String appName) {
    String target = "{identity: [clusters,'" + cluster + "']}";
    return deployUsingRest(host, port, userName, password, target, archivePath, hostHeader, appName);
    }
 public static ExecResult deployToServerUsingRest(String host, String port,
            String userName, String password, String server,
            Path archivePath, String hostHeader, String appName) {
    String target = "{identity: [servers,'" + server + "']}";
    return deployUsingRest(host, port, userName, password, target, archivePath, hostHeader, appName);
  }

Comment (vanajamukkara) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DeployUtil.java:242]: looks good, pls update with latest jenkins results when you them

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioDomainInImage.java [hunk 3]
```diff
@@ -217,7 +217,7 @@ public void testIstioDomainHomeInImage() {
 
     Path archivePath = Paths.get(ITTESTS_DIR, "../src/integration-tests/apps/testwebapp.war");
     ExecResult result = null;
-    result = DeployUtil.deployUsingRest(K8S_NODEPORT_HOST, 
+    result = DeployUtil.deployToClusterUsingRest(K8S_NODEPORT_HOST,
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioDomainInImage.java:220]: We usually use static import. Static import DeployUtil.deployToClusterUsingRest and change this DeployUtil.deployToClusterUsingRest to deployToClusterUsingRest.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioDomainInPV.java [hunk 4]
```diff
@@ -315,13 +315,13 @@ public void testIstioDomainHomeInPv() {
 
     Path archivePath = Paths.get(ITTESTS_DIR, "../src/integration-tests/apps/testwebapp.war");
     ExecResult result = null;
-    result = DeployUtil.deployUsingRest(K8S_NODEPORT_HOST, 
+    result = DeployUtil.deployToClusterUsingRest(K8S_NODEPORT_HOST,
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioDomainInPV.java:318]: We usually use static import. Static import DeployUtil.deployToClusterUsingRest and change this DeployUtil.deployToClusterUsingRest to deployToClusterUsingRest.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioMiiDomain.java [hunk 5]
```diff
@@ -224,7 +224,7 @@ public void testIstioModelInImage() {
 
     Path archivePath = Paths.get(ITTESTS_DIR, "../src/integration-tests/apps/testwebapp.war");
     ExecResult result = null;
-    result = DeployUtil.deployUsingRest(K8S_NODEPORT_HOST, 
+    result = DeployUtil.deployToClusterUsingRest(K8S_NODEPORT_HOST,
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioMiiDomain.java:227]: We usually use static import. Static import DeployUtil.deployToClusterUsingRest and change this DeployUtil.deployToClusterUsingRest to deployToClusterUsingRest.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioTwoDomainsInImage.java [hunk 6]
```diff
@@ -280,7 +280,7 @@ public void testIstioTwoDomainsWithSingleIngress() {
     logger.info("WebLogic console on domain1 is accessible");
     Path archivePath = Paths.get(ITTESTS_DIR, "../src/integration-tests/apps/testwebapp.war");
     ExecResult result = null;
-    result = DeployUtil.deployUsingRest(K8S_NODEPORT_HOST, 
+    result = DeployUtil.deployToClusterUsingRest(K8S_NODEPORT_HOST,
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioTwoDomainsInImage.java:283]: We usually use static import. Static import DeployUtil.deployToClusterUsingRest and change this DeployUtil.deployToClusterUsingRest to deployToClusterUsingRest.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioTwoDomainsInImage.java [hunk 7]
```diff
@@ -296,7 +296,7 @@ public void testIstioTwoDomainsWithSingleIngress() {
     checkConsole = checkAppUsingHostHeader(consoleUrl, domainNamespace2 + ".org");
     assertTrue(checkConsole, "Failed to access domain2 WebLogic console");
     logger.info("WebLogic console on domain2 is accessible");
-    result = DeployUtil.deployUsingRest(K8S_NODEPORT_HOST, 
+    result = DeployUtil.deployToClusterUsingRest(K8S_NODEPORT_HOST,
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioTwoDomainsInImage.java:299]: We usually use static import. Static import DeployUtil.deployToClusterUsingRest and change this DeployUtil.deployToClusterUsingRest to deployToClusterUsingRest.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java [hunk 8]
```diff
@@ -1946,10 +1946,11 @@ private static void deployApplication(String namespace, String domainUid, String
     assertNotEquals(-1, serviceNodePort, "admin server default node port is not valid");
     logger.info("Deploying application {0} to domain {1} cluster target cluster-1 in namespace {2}",
         clusterViewAppPath, domainUid, namespace);
+    String targets = "{ identity: [ clusters, 'cluster-1' ] }";
     ExecResult result = DeployUtil.deployUsingRest(K8S_NODEPORT_HOST,
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java:1950]: We usually use static import. Static import DeployUtil.deployToClusterUsingRest and change this DeployUtil.deployToClusterUsingRest to deployToClusterUsingRest.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioDomainInImage.java (hunk 3 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioDomainInImage.java]: Done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioDomainInPV.java (hunk 4 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioDomainInPV.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioTwoDomainsInImage.java (hunk 6 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioTwoDomainsInImage.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioTwoDomainsInImage.java (hunk 7 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioTwoDomainsInImage.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java (hunk 8 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTwoDomainsLoadBalancers.java:1950]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioMiiDomain.java (hunk 5 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioMiiDomain.java]: done

**Review by vanajamukkara — APPROVED** [review]

**Pull request merged**